### PR TITLE
[15.0][MIG] sale_coupon_domain_free_product: Migration to version 15.0

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
cc @Tecnativa TT38307

@chienandalu @pedrobaeza please review!

### CODE REVIEW

**Comment by stefan-tecnativa** [comment]
I am also proceeding to split the module

File: coupon_domain_free_product/tests/test_coupon_free_product_domain.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        # Every two we'll fulfill the condition
+        coupon_program_form.rule_min_quantity = 2
+        coupon_program_form.rule_products_domain = "[('id', 'in', (%s, %s))]" % (
+            cls.product_1.id,
+            cls.product_2.id,
+        )
+        coupon_program_form.reward_product_quantity = 1
+        cls.coupon_program = coupon_program_form.save()
+        # We'll be using this sale order
+        sale_form = Form(cls.env["sale.order"])
```
Comment (chienandalu) [inline coupon_domain_free_product/tests/test_coupon_free_product_domain.py]: This test case can't rely on sale.order, declare that data in the sale scoped module

Comment (stefan-tecnativa) [reply coupon_domain_free_product/tests/test_coupon_free_product_domain.py]: Understood, thanks!

**Review by chienandalu — APPROVED** [review]
LGTM :+1:

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**